embedded struct pointer panic

I always get a panic when my struct has the similar structure as below, here's a minimal example

```go
package main

import (
	"github.com/leebenson/conform"
	"log"
)

type Child struct {
	ChildName string `conform:"trim"`
	Numbers   []uint64
}

type Parent struct {
	Child *Child
}

func main() {
	child := Child{
		ChildName: "   child   ",
		Numbers:   []uint64{1, 2, 3},
	}
	parent := Parent{
		Child: &child,
	}
	if err := conform.Strings(&parent); err != nil {
		log.Println(err)
	} else {
		log.Println(*parent.Child)
	}
}
```

output:

```text
panic: reflect.Value.Convert: value of type string cannot be converted to type uint64
```

possible fix

`conform.go:243`

```go
if elType.ConvertibleTo(reflect.TypeOf(str)) || elType.ConvertibleTo(reflect.TypeOf(&str)) {
```

change to

```go
if elType == reflect.TypeOf(str) || elType == reflect.TypeOf(&str) {
```

output:

```text
2022/12/06 22:11:56 {child [1 2 3]}
```

I might try to make a pr to fix it later, any help or explanation is appreciated, thanks in advance!

--- a/conform.go
+++ b/conform.go
@@ -243,6 +243,9 @@ func Strings(iface interface{}) error {
 				if (elType.ConvertibleTo(reflect.TypeOf(str)) && reflect.TypeOf(str).ConvertibleTo(elType)) ||
 					(elType.ConvertibleTo(reflect.TypeOf(&str)) && reflect.TypeOf(&str).ConvertibleTo(elType) ) {
 					tags := v.Tag.Get("conform")
+					if len(tags) <= 0 {
+						continue
+					}
 					for i := 0; i < el.Len(); i++ {
 						el.Index(i).Set(transformValue(tags, el.Index(i)))
 					}

